### PR TITLE
feat: reset pasword backend

### LINKED ISSUE #108: [FEATURE]: Forgot password
## Summary

- Generate recovery password token
- Send recovery password link to the users email 
- Reset password model and schema 

This feature should follow the [forgot password flow](https://www.figma.com/file/zhPJunvjgLcWD2bklt4uG0/User-Flows-%2B-User-Roles%2FPermissions?node-id=183%3A1346)

## User story it relates to

#106

## Acceptance Criteria

- should be created a reset password module:
   - reset password service
   - recover password method should generate a 15 min token to allow password changes and send the reset link
   - resent reset link method should regenerate a token and send again the reset link to the user 
 
- Auth controller should have `/auth/recoverPassword` endpoint that invokes the respective method in the previous mentioned module.

- Should follow the figma flow

### PR DESCRIPTION
**Pull request by r-dmatos**
Relates to  #108 

Reset password,
user is given a token via email, 
there is an verification to find out if the token corresponds to the user,
user changes the password,
there is a verification to confirm the passwords given are the same.

this pr closes #108

### CODE REVIEW

**Comment by dsousa12** [comment]
Please remove `public/icons`. Icons now is used with `<Icon />` component, where you need to pass `name` as prop.
The name it's the `id=""` in each `<symbol>` tag inside `Sprite.tsx`.

**Comment by dsousa12** [comment]
Please remove `.vscode/` folder inside `frontend`.

**Comment by dsousa12** [comment]
Please delete `package-lock.json` on root folder

**Review by dsousa12 — CHANGES_REQUESTED** [review]

File: frontend/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (dsousa12) [inline frontend/.vscode/launch.json:1]: Delete his file.
Don't make any sense

File: frontend/public/icons/envelope.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<svg width="55" height="52" viewBox="0 0 55 52" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/envelope.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with `<Icon />` component

File: frontend/public/icons/eye.svg [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/eye.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with <Icon /> component

File: frontend/public/icons/gitHub.svg [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<svg viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/gitHub.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with <Icon /> component

File: frontend/public/icons/info.svg [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/info.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with <Icon /> component

File: frontend/public/icons/leftArrow.svg [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/leftArrow.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with <Icon /> component

File: frontend/public/icons/microsoft.svg [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/microsoft.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with <Icon /> component

File: frontend/public/icons/toast/alert.svg [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<svg width="31" height="32" viewBox="0 0 31 32" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (dsousa12) [inline frontend/public/icons/toast/alert.svg:1]: Delete the icon
If this icon does not exist on Sprite, please add it and use with <Icon /> component

File: backend/package.json [hunk 9]
```diff
@@ -48,7 +48,9 @@
 		"eslint-config-airbnb-typescript": "^17.0.0",
 		"express": "^4.17.3",
 		"joi": "^17.6.0",
+		"jsonwebtoken": "^8.5.1",
 		"jwt-decode": "^3.1.2",
```
Comment (dsousa12) [inline backend/package.json:52]: I don't have 100% sure, but I think you don't need a package to decode and encode.
Maybe you can use something like this: https://github.com/nestjs/jwt
But I will explore more about this topic

File: frontend/schema/schemaResetPasswordForm.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import * as z from 'zod';
+
+const SchemaResetPasswordForm = z
+	.object({
+		newPassword: z
+			.string()
+			.nonempty('Please enter your password.')
+			.regex(/.*[A-Z].*/, 'One uppercase character')
```
Comment (dsousa12) [inline frontend/schema/schemaResetPasswordForm.tsx:8]: ### Suggestion
Change this multiples regexs on this schema for:
```
z 
  .string()
  .nonempty('Please enter your password.')
  .regex(
        /^(?=.*[A-Za-z])(?=.*\d)(?=.*\W)[A-Za-z\d\W]{8,}$/,
        'Use at least 8 characters, upper and lower case letters, numbers and symbols like !”?$%^&).'
  ),
```
Basically that regex validate if have the leght of 8 chars, containing uppercase letter(s), lowecase letter(s), number(s) and special char(s).

File: frontend/components/auth/LoginForm/index.tsx [hunk 11]
```diff
@@ -135,7 +135,7 @@ const LoginForm: React.FC<LoginFormProps> = ({ setShowTroubleLogin }) => {
 					/>
 
 					<LoginButton type="submit" disabled={loading.credentials} size="lg">
-						{loading.credentials && <DotsLoading color="primary800" size={10} />}
```
Comment (dsousa12) [inline frontend/components/auth/LoginForm/index.tsx:138]: Why you remove the `<DotsLoading>` component?

**Review by dsousa12 — CHANGES_REQUESTED** [review]

File: frontend/schema/schemaResetPasswordForm.tsx (hunk 10 above)
Comment (r-dmatos) [reply frontend/schema/schemaResetPasswordForm.tsx:8]: thank you for the suggestion. uploading changes now

**Pull request merged**